src/Adapter/Language/LanguageImageManager.php

```diff
@@ -0,0 +1,204 @@
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Language;
+
+use ImageType;
+
+/**
+ * Handles language images (flag, "no image" placeholders)
+ */
+class LanguageImageManager
+{
+    /**
+     * Path where images are saved to
+     */
+    const IMG_PATH = _PS_IMG_DIR_ . '/l/';
+
+    /**
+     * Path where flags are stored
+     */
+    const FLAGS_SOURCE = _PS_BO_ALL_THEMES_DIR_ . 'new-theme/static/flags/%s.jpg';
+
+    /**
+     * Path where flags are copied to
+     */
+    const FLAGS_DESTINATION = self::IMG_PATH . "%d.jpg";
+
+    /**
+     * Default flag
+     */
+    const FALLBACK_FLAG_SOURCE = self::IMG_PATH . 'none.jpg';
+
+    const IMAGE_DIRECTORIES = [
+        _PS_CAT_IMG_DIR_,
+        _PS_MANU_IMG_DIR_,
+        _PS_PROD_IMG_DIR_,
+        _PS_SUPP_IMG_DIR_,
+    ];
+
+    const PLACEHOLDER_IMAGE_NAME_PATTERNS = [
+        '%s.jpg',
+        '%s-default-%s.jpg'
+    ];
+
+    const DEFAULT_LANGUAGE_CODE = 'en';
+
+    /**
+     * Sets up the language flag image for the given language
+     *
+     * @param string $localeCode IETF language tag
+     * @param int $langId Language id
+     * @param string|null $flagCode If provided, use this flag code. By default, auto-detect using locale code.
+     */
+    public function setupLanguageFlag(string $localeCode, int $langId, ?string $flagCode = null): void
+    {
+        $flagCode = $flagCode ?? $this->getFlagCountryCodeFromLocale($localeCode);
+
+        $flagPath = $this->getFlagPath($flagCode);
+
+        if (!file_exists($flagPath)) {
+            $flagPath = static::FALLBACK_FLAG_SOURCE;
+        }
+
+        $destinationPath = $this->getFlagDestination($langId);
+
+        $this->unlinkIfExists($destinationPath);
+
+        copy($flagPath, $destinationPath);
+    }
+
+    /**
+     * Creates default copies for the "no image" image
+     *
+     * @param string $isoCode 2-letter ISO code
+     */
+    public function setupDefaultImagePlaceholder(string $isoCode): void
+    {
+        $filesToCopy = [
+            $this->getPlaceholderImageFilename(static::DEFAULT_LANGUAGE_CODE) => $this->getPlaceholderImageFilename($isoCode)
+        ];
+
+        $imageTypes = ImageType::getAll();
+        if (!empty($imageTypes)) {
+            foreach (array_keys($imageTypes) as $alias) {
+                $formattedImageType = ImageType::getFormattedName($alias);
+                $from = $this->getPlaceholderImageFilename(static::DEFAULT_LANGUAGE_CODE, $formattedImageType);
+                $to = $this->getPlaceholderImageFilename($isoCode, $formattedImageType);
+                $filesToCopy[$from] = $to;
+            }
+        }
+
+        foreach (static::IMAGE_DIRECTORIES as $destinationDir) {
+            foreach ($filesToCopy as $sourceFile => $newFile) {
+                @copy(static::IMG_PATH . $sourceFile, $destinationDir . $newFile);
```

Comment:
Why adding `@` here and not on line 92?

Reply:
No particular reason, I just kept it pretty much as it was.
I actually have the feeling that this piece of code (which are basically the thumbnails for the "no image" image) belongs elsewhere, but I didn't want to stray even farther from my original intention.